osu.Game.Modes.Taiko/Objects/TaikoHitObject.cs

```diff
@@ -0,0 +1,54 @@
+﻿using osu.Game.Modes.Objects;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Samples;
+using osu.Game.Beatmaps.Timing;
+
+namespace osu.Game.Modes.Taiko.Objects
+{
+    public class TaikoHitObject : HitObject
+    {
+        public double HitWindowGreat;
+        public double HitWindowGood;
+        public double HitWindowMiss;
+
+        public double PreEmpt;
+        public float Scale = 1;
+        public bool Kiai;
+
+        public TaikoHitType Type => ((Sample?.Type ?? SampleType.None) & (~SampleType.Finish & ~SampleType.Normal)) == 0 ? TaikoHitType.Don : TaikoHitType.Katsu;
+        public bool IsFinisher => ((Sample?.Type ?? SampleType.None) & SampleType.Finish) > 0;
+
+        public override void SetDefaultsFromBeatmap(Beatmap beatmap)
+        {
+            base.SetDefaultsFromBeatmap(beatmap);
+
+            // Don't ask... Old osu! had a random multiplier here, that we now have to multiply everywhere
+            float fudgeFactor = 1.4f;

```

Comment:
You can make all these fudge factors const. Might also want to share them across multiple files. Seems weird to locally declare them in multiple places.